src/org/rascalmpl/uri/URIUtil.java

```diff
@@ -182,21 +182,11 @@ private static ISourceLocation createLocation(String scheme, String authority,
 			String path) throws URISyntaxException {
 		return vf.sourceLocation(scheme, authority, path);
 	}
-
-	private static final URI invalidURI = URI.create("unknown:///");
 	
-	/**
-	 * Returns an URI which cannot be read/write to.
-	 * @return
-	 */
-	public static URI invalidURI() {
-		return invalidURI;
-	}
-	
-	private static final ISourceLocation invalidLocation = vf.sourceLocation(invalidURI);
+	private static final ISourceLocation unknownLocation = rootLocation("unknown");
 	
-	public static ISourceLocation invalidLocation() {
-		return invalidLocation;
+	public static ISourceLocation unknownLocation() {
```

Comment:
this renaming does create more binary incompatibility than needed. Specifically in places that shouldn't have to change for the new functionality introduced.

can we keep the old one around, and put a `@deprecated` on there? just so that down-stream projects can just keep working without needing to update this?

(same is true for the `INTERPRETER` enum typofix)

Reply:
I think there is so much damage in this PR that it's better to bite the bullet and simply make the best of it.